### PR TITLE
xcode install changes

### PR DESCRIPTION
**Pull request by odra**
# Changes

* installs xcode as sudo user.
* removes rvm from installation process.
* Handles apple developer portal authentication error.
* Fixes Xcode.app symlink  by adding a `default_xcode_version`  var that defaults to the first list item of `xcode_version`
* Renamed pod_update task to  "cocoapods.yml" as it installs and runs `pod repo update`

Ideally it should be tested in a clean osx environment with no rvm, no xcode, not homebrew and so on.

### CODE REVIEW

File: provision-osx/tasks/install_xcode.yml [hunk 1]
```diff
@@ -55,14 +62,40 @@
...
   failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
   changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
+  become: yes
   environment:
     XCODE_INSTALL_USER: "{{ xcode_install_user }}"
     XCODE_INSTALL_PASSWORD: "{{ xcode_install_password }}"
+
+-
+  name: Check xcode output
+  debug: msg="Checking {{ item.cmd }} output"
```
Comment (odra) [inline provision-osx/tasks/install_xcode.yml]: I am not sure if it is the best way to validate multiple shell outputs, open to suggestions

File: provision-osx/tasks/install_xcode.yml [hunk 2]
```diff
@@ -55,14 +62,40 @@
...
+    path: /Applications/Xcode.app
+  register: xcode_symlink
+
+-
+  name: Set default xcode symlink path
+  set_fact:
+    xcode_symlink_path: "/Applications/Xcode-{{ xcode_default_version }}.app"
+
+-
+  name: Set default xcode symlink
```
Comment (odra) [inline provision-osx/tasks/install_xcode.yml]: I noticed that the `xcversion` command will set a symlink to the last installed xcode  version which I don't think is the ideal behavior.

**Comment by PhilipGough** [comment]
:eyes will review and verify

File: provision-osx/tasks/install_xcode.yml (hunk 1 above)
Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: So here we are checking all installs to ensure the credentials are valid is that correct? Won't the preceding task fail anyway if that is the case?

File: provision-osx/tasks/install_xcode.yml (hunk 2 above)
Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: We are creating a symlink here with the first version in the list is that correct? If so I think this is fine but we need to document it or allow them to set the default might be preferably, what do you think? We can always fall back on first in list

Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: Ignore. Closer inspection this is done

File: provision-osx/tasks/install_xcode.yml [hunk 3]
```diff
@@ -55,14 +62,40 @@
...
   name: Install xcode versions
-  shell: "source ~/.bash_profile && xcversion install {{ item }}"
+  shell: "{{ xcversion_bin.stdout }} install {{ item }}"
   args:
     executable: /bin/bash
   with_items: "{{ xcode_versions }}"
   when: xcode_install_user is defined and xcode_install_password is defined
   register: xcode_install
   failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
   changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
```
Comment (matskiv) [inline provision-osx/tasks/install_xcode.yml]: btw: in two conditions above we should be using `xcode_install` variable. @odra if you agree, could you please add a fix into this PR as well ? Thanks.

Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: Good spot. I'll push this fix

File: provision-osx/tasks/install_xcode.yml (hunk 1 above)
Comment (odra) [inline provision-osx/tasks/install_xcode.yml]: it was not failing when the problem was developer portal authentication

Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: Seems fine then or we can alter the `failed_when` conditionon the previous task but i have no problem with this approach for now

Comment (odra) [inline provision-osx/tasks/install_xcode.yml]: That was my thought as well, but that check if quite big already so I preffered to move it to its own task step

File: provision-osx/tasks/cocoapods.yml [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
+
+-
   name: Update Cocoapods
   shell: bash -lc "pod repo update"
   register: pod_repo_update
   changed_when: pod_repo_update.stdout.find('Already up-to-date') == -1
   environment:
     PATH: "{{ buildfarm_env_path.stdout }}"
-    GEM_HOME: "{{ buildfarm_env_gem.stdout }}"
     LANG: "{{ buildfarm_lang_env_var.stdout }}"
```
Comment (matskiv) [inline provision-osx/tasks/cocoapods.yml]: I think this should be just `buildfarm_env_lang` without .stdout OR it should be `buildfarm_env_lang`

**Comment by odra** [comment]
@PhilipGough I bumped xcode-installl version since the upstream repository got merged

**Comment by odra** [comment]
I also saw some bugs with the native gem module so I changed it to use the shell one again.

File: provision-osx/tasks/install_xcode.yml [hunk 5]
```diff
@@ -1,68 +1,91 @@
...
-  when: xcode_install_user is defined and xcode_install_password is defined
-  failed_when: xcode_cli_tools_install.rc != 0 and xcode_cli_tools_install.stdout.find('already installed') == -1
-  changed_when: xcode_cli_tools_install.stdout.find('already installed') == -1
+  name: Update cached xcode versions for {{ ansible_user }}
+  command: "{{ xcversion_bin.stdout }} update"
+  become: yes
   environment:
     XCODE_INSTALL_USER: "{{ xcode_install_user }}"
     XCODE_INSTALL_PASSWORD: "{{ xcode_install_password }}"
-    FASTLANE_SESSION: "{{ xcode_install_session_token | default(omit) }}"
```
Comment (PhilipGough) [inline provision-osx/tasks/install_xcode.yml]: Should this be added back in where required as I see we have it in our current docs. Ping @odra

Comment (odra) [reply provision-osx/tasks/install_xcode.yml]: yeah let me add it back

**Pull request merged**